Source/WebKit/Shared/unix/AuxiliaryProcessMain.cpp

```diff
@@ -45,18 +47,44 @@ AuxiliaryProcessMainCommon::AuxiliaryProcessMainCommon()
 #endif
 }
 
+// The command line is constructed in ProcessLauncher::launchProcess.
 bool AuxiliaryProcessMainCommon::parseCommandLine(int argc, char** argv)
 {
-    ASSERT(argc >= 3);
-    if (argc < 3)
+#if OS(LINUX)
+    int minimumNumArgs = 4;
+#else
+    int minimumNumArgs = 3;
+#endif
+
+#if USE(LIBWPE) && !ENABLE(BUBBLEWRAP_SANDBOX)
+    if (ProcessProviderLibWPE::singleton().isEnabled())
+        minimumNumArgs = 3;
+#endif
+
+    if (argc < minimumNumArgs)
         return false;
 
     m_parameters.processIdentifier = ObjectIdentifier<WebCore::ProcessIdentifierType>(atoll(argv[1]));
     m_parameters.connectionIdentifier = IPC::Connection::Identifier { atoi(argv[2]) };
+    if (!m_parameters.processIdentifier || !m_parameters.connectionIdentifier)
+        return false;
+
+#if OS(LINUX)
+    if (minimumNumArgs == 4) {
+        int pidSocket = atoi(argv[3]);
+        if (!pidSocket)
+            return false;
```

Comment:
We have `WTF::parseInteger()`, which does better checking than `atoi`:

```suggestion
        auto pidSocket = parseInteger<int>(argv[3]);
        if (!pidSocket || *pidSocket < 0)
            return false;
```

Reply:
Good point, will change that before landing.

Reply:
Done.